pkg/controller/controller_utils.go

```diff
@@ -629,6 +686,135 @@ func maxContainerRestarts(pod *api.Pod) int {
 	return maxRestarts
 }
 
+type podControllerRefManager struct {
+	podControl         PodControlInterface
+	controllerObject   api.ObjectMeta
+	controllerSelector labels.Selector
+	controllerKind     unversioned.GroupVersionKind
+}
+
+// NewPodControllerRefManager returns a podControllerRefManager that exposes
+// methods to manage the controllerRef of pods.
+func NewPodControllerRefManager(
+	podControl PodControlInterface,
+	controllerObject api.ObjectMeta,
+	controllerSelector labels.Selector,
+	controllerKind unversioned.GroupVersionKind,
+) *podControllerRefManager {
```

Comment:
I think @krousey convinced me that the pattern of returning a private struct isn't great. I suggest making this struct public, but leave its members private.